tests/h/views/accounts_test.py

```diff
@@ -660,6 +660,29 @@ def test_get_sets_subscriptions_data_in_form(self,
             'notifications': set(['reply']),
         })
 
+    def test_get_for_user_without_email_address(self,
+                                                form_validating_to,
+                                                pyramid_request):
+        controller = views.NotificationsController(pyramid_request)
+        controller.form = form_validating_to({})
+
+        result = controller.get()
+
+        assert 'form' not in result
+        assert result['user_has_email_address'] is None
+
+    def test_get_for_user_with_email_address(self,
+                                             form_validating_to,
+                                             pyramid_request):
+        controller = views.NotificationsController(pyramid_request)
+        controller.form = form_validating_to({})
+        pyramid_request.user = mock.Mock(username='janedoe')
```

Comment:
A `mock` here isn't invalid per se, but our testing conventions lead us to not using bare mocks (except in atypical situations).  This test would be more robust (and realistic) if it used `factories.User()`-generated model here instead of a mock. That's our usual approach in tests like this. It will help us catch more little oddities...sound OK?

Reply:
Yeah, makes sense. We seem to be using it in other tests, so I thought that might be ok.